Update to react 17

I don't think that description is needed.

I don't mind to do, just update me if you allow.

Thanks.

I believe I've run into the same issue here. As suggested in the error message, it does work fine with --legacy-peer-deps, so it's not critical.

Steps to reproduce: run `npm install --save @koale/useworker` in a project which uses React 17.0.2
Expected result: package installs
Actual result: receive the following error:
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: react-client@0.1.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from @koale/useworker@4.0.2
npm ERR! node_modules/@koale/useworker
npm ERR!   @koale/useworker@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/cassm/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/cassm/.npm/_logs/2022-03-09T12_41_04_040Z-debug-0.log

The mentioned log files are attached.
[eresolve-report.txt](https://github.com/alewin/useWorker/files/8214765/eresolve-report.txt)
[2022-03-09T12_41_04_040Z-debug-0.log](https://github.com/alewin/useWorker/files/8214769/2022-03-09T12_41_04_040Z-debug-0.log)

Same issue with react 18:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: test-typescript-web-wrk-canvas@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from @koale/useworker@4.0.2
npm ERR! node_modules/@koale/useworker
npm ERR!   @koale/useworker@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/vittorio/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vittorio/.npm/_logs/2022-04-09T14_41_58_833Z-debug-0.log

```
however using `--legacy-peer-deps` seems to workaround the problem

--- a/packages/useWorker/package.json
+++ b/packages/useWorker/package.json
@@ -15,7 +15,7 @@
     "test": "npm run build && npm run test:unit"
   },
   "peerDependencies": {
-    "react": ">=16.0.0 <20.0.0"
+    "react": ">=16.8.0 <20.0.0"
   },
   "devDependencies": {
     "@babel/plugin-proposal-optional-chaining": "^7.9.0",

--- a/packages/useWorker/package.json
+++ b/packages/useWorker/package.json
@@ -15,7 +15,7 @@
     "test": "npm run build && npm run test:unit"
   },
   "peerDependencies": {
-    "react": "^16.8.0"
+    "react": ">=16.0.0 <20.0.0"
   },
   "devDependencies": {
     "@babel/plugin-proposal-optional-chaining": "^7.9.0",

